src/components/mortgage-investment-compare/index.js

```diff
@@ -0,0 +1,220 @@
+      return FV(
+        (state.investmentRate - state.inflation) / 100 / COMPOUND_FREQUENCY,
+        option1.term * COMPOUND_FREQUENCY,
+        option2.pmt - option1.pmt,
+        0
+      )
+    } else if (option1.term < option2.term) {
+      return FV(
+        (state.investmentRate - state.inflation) / 100 / COMPOUND_FREQUENCY,
+        (option2.term - option1.term) * COMPOUND_FREQUENCY,
+        option1.pmt,
+        0
+      )
+    }
+  }
+
+  // Returns opportunity cost of choosing option1 over option2. Opportunity cost can be positive and negative.
+  calculateOpportunityCost (option1, option2) {
+    return Math.abs(option1.fv - option2.fv)
+  }
+
+  // Runs every time an input is updated and uses input's name tag to make specific changes in the state
+  updateInput (e) {
+    let state = Object.assign({}, this.state)
+    const key = e.target.name
+
+    const value = parseFloat(e.target.value)
+
+    _.set(state, key, value)
+    if (!isNaN(value)) {
+      state = this.calculateAll(state)
+    }
+    this.setState(state)
+  }
+
+  // Runs all calculations and returns a modified state
+  calculateAll (state) {
+    option1 = state.options.option1
+    option2 = state.options.option2
+
+    option1.pmt = this.calculatePMT(option1, state)
+    option2.pmt = this.calculatePMT(option2, state)
+
+    option1.interestAmt = this.calculateInterestAmt(option1, state)
+    option2.interestAmt = this.calculateInterestAmt(option2, state)
+
+    option1.fv = this.calculateFV(option1, option2, state)
+    option2.fv = this.calculateFV(option2, option1, state)
+
+    state.optCost = this.calculateOpportunityCost(option1, option2)
+
+    const [shorter, longer] = _.sortBy([option1, option2], 'term')
+    state.shorterOption = shorter.term === option1.term ? 'option1' : 'option2'
+    state.longerOption = longer.term === option1.term ? 'option1' : 'option2'
+
+    state.yearlyResultsByOption = getYearly({
+      loanAmt: state.loanAmt,
+      investmentRate: state.investmentRate / 100,
+      inflationRate: state.inflation / 100,
+      shorterOption: {
+        mortgageRate: shorter.interestRate / 100,
+        mortgageTerm: shorter.term
+      },
+      longerOption: {
+        mortgageRate: longer.interestRate / 100,
+        mortgageTerm: longer.term
+      }
+    })
+
+    return state
+  }
+
+  render () {
+    const shorterOption = this.state.options[this.state.shorterOption]
+    const longerOption = this.state.options[this.state.longerOption]
+    const bestOption = _.maxBy([shorterOption, longerOption], 'fv')
+
+    return (
+      <Row>
+        <Col xs='4'>
+          <InputCard
+            loanAmt={this.state.loanAmt}
+            inflation={this.state.inflation}
+            investmentRate={this.state.investmentRate}
+            updateInput={this.updateInput}
+            option1={option1}
+            option2={option2}
+            shorterOption={shorterOption}
+            longerOption={longerOption}
```

Comment:
sug: I'm a bit concerned that this is a double-bind, since people can modify the input props. What does React recommend when child components modify data?

Outside of React, I'm used to the child emitting events that propagate data back to the parent

Reply:
Can you show me an example of how this would be problematic? I've seen huge React applications full of child components that modify the parent's data. Never encountered issues and don't think it's a bad practice. The truth is that the child isn't really modifying the data because you're still pasing `updateInput` which runs within the parent.

Reply:
moving the state to the child and extracting it for use in the parent will be a headache without redux if that's what you're thinking about

Reply:
Looking at some online examples, I think you can have the component have its own state to modify. If it makes changes, we can call an event handler that returns the whole dataset in the component.

The parent can the consume that in `calculateAll`

Reply:
yup that is the headache I was talking about and the event handler returning a whole dataset behavior sounds a lot like what redux does. In the hundreds of components I've seen and built, I have never seen this done. I have even gone through the troubles of moving all of a child's state to a parent's state in some situations to avoid doing this. I still don't understand how the current implementation can be problematic.